cypress-tests/cypress/support/pageBuilder/pbDeleteAllTemplates.ts

```diff
@@ -0,0 +1,70 @@
+import { gqlClient } from "../utils";
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface Chainable {
+            // Combined method to list and delete all templates
+            pbDeleteAllTemplates(): void;
+        }
+    }
+}
+
+const LIST_QUERY = /* GraphQL */ `
+    query ListPageTemplates {
+        pageBuilder {
+            listPageTemplates {
+                data {
+                    id
+                }
+            }
+        }
+    }
+`;
+
+const DELETE_MUTATION = /* GraphQL */ `
+    mutation DeletePageTemplate($id: ID!) {
+        pageBuilder {
+            deletePageTemplate(id: $id) {
+                error {
+                    code
+                    message
+                }
+            }
+        }
+    }
+`;
+
+Cypress.Commands.add("pbDeleteAllTemplates", () => {
+    cy.login().then(user => {
+        // First, list all page templates
+        gqlClient
+            .request({
+                query: LIST_QUERY,
+                authToken: user.idToken.jwtToken
+            })
+            .then(listResponse => {
+                const templates = listResponse.pageBuilder.listPageTemplates.data;
+
+                // Loop through the templates and delete each one
+                templates.forEach((template: { id: any }) => {
+                    gqlClient
+                        .request({
+                            query: DELETE_MUTATION,
+                            variables: {
+                                id: template.id
+                            },
+                            authToken: user.idToken.jwtToken
+                        })
+                        .then(deleteResponse => {
+                            const error = deleteResponse.pageBuilder.deletePageTemplate.error;
+                            if (error) {
+                                // Handle any errors that occur during deletion if needed
+                                // You can use Cypress log or assertion to report the error
+                                cy.log(`Error deleting template with ID: ${template.id}`);
+                            }
+                        });
+                });
+            });
```

Comment:
```suggestion
        await Promise.all(
            templates.map((template: { id: any }) => {
                return gqlClient
                    .request({
                        query: DELETE_MUTATION,
                        variables: {
                            id: template.id
                        },
                        authToken: user.idToken.jwtToken
                    })
                    .then(deleteResponse => {
                        const error = deleteResponse.pageBuilder.deletePageTemplate.error;
                        if (error) {
                            // Handle any errors that occur during deletion if needed
                            // You can use Cypress log or assertion to report the error
                            cy.log(`Error deleting template with ID: ${template.id}`);
                        }
                    });
            })
        );
```

Reply:
I did a change above because not only the delete ops will be executed in parallel, but also, the function will return a promise which will be resolved once all of the ops have been resolved. In other words, the next command will be executed only once these ops have been all completed.